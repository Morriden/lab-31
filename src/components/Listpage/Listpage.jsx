import React from 'react';
import PropTypes from 'prop-types';

const Listpage = ({ lists }) => {
  const listElements = lists.map(list => (
    <li key = {list}>
      {list}
    </li>
  ));

  return (
    <ul>
      {listElements}
    </ul>
  );

};
Listpage.propTypes = {
  lists: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default Listpage;
