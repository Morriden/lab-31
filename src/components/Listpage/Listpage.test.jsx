import React from 'react';
import { shallow } from 'enzyme';
import Listpage from './Listpage';

describe('List component', () => {
  it('should show a list', () => {
    const wrapper = shallow(
      <Listpage lists={['test 1', 'test 2', 'test 3']}
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
