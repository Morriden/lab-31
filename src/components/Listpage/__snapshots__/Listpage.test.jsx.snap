// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should show a list 1`] = `
<ul>
  <li
    key="test 1"
  >
    test 1
  </li>
  <li
    key="test 2"
  >
    test 2
  </li>
  <li
    key="test 3"
  >
    test 3
  </li>
</ul>
`;
